File: 441/arranging-coins.go
func arrangeCoins(n int) int {
	if n == 0 {
		return 0
	} else {
		for k := 1; ; k++ {
			if n >= k*(k+1)/2 && n < (k+1)*(k+2)/2 {
				return k
			}
		}
	}
}

File: 441/arranging-coins.js
/**
 * @param {number} n
 * @return {number}
 */
var arrangeCoins = function (n) {
    if (!n) return 0;
    for (let k = 1; ; k++) {
        if (n >= k * (k + 1) / 2 && n < (k + 1) * (k + 2) / 2)
            return k;
    }
};

var arrangeCoins = function(n) {
    let ans = Math.sqrt(1 + 8 * n) - 1;
    ans /= 2;
    return ~~ans;
}